adapta/storage/distributed_object_store/v3/datastax_astra/_model_mappers.py

```diff
@@ -0,0 +1,309 @@
+"""Model mapper module"""
+import datetime
+import enum
+import sys
+import typing
+from abc import ABC, abstractmethod
+from dataclasses import is_dataclass, fields
+from typing import Type, Optional, List
+import re
+
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine.columns import Column
+from cassandra.cqlengine import columns
+
+TModel = typing.TypeVar("TModel")  # pylint: disable=C0103
+
+
+class CassandraModelMapper(ABC):
+    """
+    Abstract class for mapping various data models to Cassandra models.
+
+        :param: keyspace: Optional keyspace name, if not provided in the client constructor.
+        :param: table_name: Astra table name, if it cannot be inferred from class name by converting it to snake_case.
+        :param: primary_keys: An optional list of columns that constitute a primary key.
+        :param: partition_keys: An optional list of columns that constitute a partition key.
+        :param: custom_indexes: An optional list of columns that have a custom index on them.
+        :param: select_columns: An optional list of columns to select from the entity.
```

Comment:
Isn't this a pos-process for read/filters only? I think we map models 1-1 by default?